sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -200,28 +210,32 @@ case class TakeOrderedAndProjectExec(
   protected override def doExecute(): RDD[InternalRow] = {
     val ord = new LazilyGeneratedOrdering(sortOrder, child.output)
     val childRDD = child.execute()
-    val singlePartitionRDD = if (childRDD.getNumPartitions > 1) {
-      val localTopK = childRDD.mapPartitions { iter =>
-        org.apache.spark.util.collection.Utils.takeOrdered(iter.map(_.copy()), limit)(ord)
-      }
-      new ShuffledRowRDD(
-        ShuffleExchangeExec.prepareShuffleDependency(
-          localTopK,
-          child.output,
-          SinglePartition,
-          serializer,
-          writeMetrics),
-        readMetrics)
+    if (childRDD.getNumPartitions == 0) {
+      sparkContext.parallelize(Array.empty[InternalRow], 1)
```

Comment:
nit: how about using `ParallelCollectionRDD` directly?

Reply:
ok, I will update it.